app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -792,11 +840,9 @@ class NewMessageViewModel @Inject constructor(
 
         subject = subjectValue
 
-        uiBody = uiBodyValue
-        uiSignature = uiSignatureLiveData.value
-        uiQuote = uiQuoteLiveData.value
-
-        body = getWholeBody()
+        val signature = uiSignatureLiveData.value
+        val quote = uiQuoteLiveData.value
+        body = getWholeBody(uiBodyValue, signature, quote)
```

Comment:
These three fields are `@Ignore` and cannot ever be saved to Realm.

These values are only ever used during snapshot comparison and during `initDraftAndViewModel()`. In this second usage, it is set at the beginning of the method (either by actually setting the objet's field or by instantiating a new instance of draft with its default value) and it is read by the end of the method.

To avoid impacting code that did not relate to the body I refrained from trying to change that part. Because I only modified the body and kept the signature and quote perfectly intact, I tried to not touch too much how things worked for signatures and quotes.

But because I changed how `saveSnapshot()` works, we could argue that I could've used this opportunity to remove uiBody, uiSignatures and uiQuotes from the Draft class.

We should probably either use variables defined inside the scope of the method or if we really need the side effect, we can have them plainly defined inside the view model and use the view model to transmit the data across parts of the `initDraftAndViewModel()`. All of this while passing in uiSignature and uiQuote as arguments of `saveSnapshot` the same way I had to do for uiBody.

What do you think of all this? Also, is this important enough that we need to refactor that part? If you agree with keeping any non-body matter intact in this PR, I could maybe create a new branch were I try and refactor that part entirely as an independent new task anyway.

Anyway, great question!